client/src/js/modules/contact/views/addcontact.js

```diff
@@ -17,26 +18,68 @@ define(['views/form',
         
         ui: {
             country: 'select[name=COUNTRY]',
+            pid: 'select[name=PERSONID]',
+            cn: 'input[name=CARDNAME]',
+            fn: 'input[name=FAMILYNAME]',
+            gn: 'input[name=GIVENNAME]',
+            pn: 'input[name=PHONENUMBER]',
+            ea: 'input[name=EMAILADDRESS]',
+            ln: 'input[name=LABNAME]',
+            ad: 'textarea[name=ADDRESS]',
+            ci: 'input[name=CITY]',
+            pc: 'input[name=POSTCODE]',
+        },
+
+        events: {
+            'change @ui.pid': 'fillUserInfo',
+        },
+
+        fillUserInfo: function() {
+            var u = this.users.findWhere({ PERSONID: this.ui.pid.val() })
+            if (u){
+                this.ui.cn.val(u.get('FULLNAME'))
+                this.ui.fn.val(u.get('FAMILYNAME'))
+                this.ui.gn.val(u.get('GIVENNAME'))
+                this.ui.pn.val(u.get('PHONENUMBER'))
+                this.ui.ea.val(u.get('EMAILADDRESS'))
+                this.ui.ln.val(u.get('LABNAME'))
+                this.ui.ad.val(u.get('ADDRESS'))
+                this.ui.ci.val(u.get('CITY'))
+                this.ui.pc.val(u.get('POSTCODE'))
+                this.ui.country.val(u.get('COUNTRY'))
+            }
         },
 
         initialize: function() {
             this.countries = new Countries()
             this.countries.state.pageSize = 9999
             this.ready = this.countries.fetch()
+            this.users = new Users()
+            this.users.queryParams.login = 1
+            this.ready2 = this.users.fetch()
```

Comment:
```suggestion
            this.countries = new Countries()
            this.countries.state.pageSize = 9999
            this.users = new Users()
```

Used this opportunity to do a bit of rewriting; `this.ready` is redundant, and we don't really need the `queryParams`, seems like the API responds just fine without it (unless you know of any weird edge case)

Reply:
The queryParams.login is to only show users with a fedid field, see https://github.com/DiamondLightSource/SynchWeb/blob/master/api/src/Controllers/UserController.php#L193 and https://github.com/DiamondLightSource/SynchWeb/blob/master/api/src/Model/Services/UserData.php#L92

Reply:
Oh, I see, I thought login mapped directly to the login column instead, I'll edit the suggestion